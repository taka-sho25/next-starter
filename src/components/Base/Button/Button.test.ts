import { cleanup, render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import React from 'react'
import { afterEach, beforeEach, describe, expect, test, vi } from 'vitest'

import { Button } from './'

let component: React.ReactNode

const onClickMock = vi.fn()

beforeEach(() => {
  component = Button({
    type: 'button',
    children: React.createElement('div', {}, 'hello'),
    onClick: onClickMock,
  })
})

afterEach(() => {
  cleanup()
})

describe('Button Component', () => {
  test('default snapshot', () => {
    const { container } = render(component)
    expect(container).toMatchSnapshot()
  })
  test('call onClick Event', async () => {
    render(component)
    expect(onClickMock).toHaveBeenCalledTimes(0)

    await userEvent.click(screen.getByRole('button'))
    expect(onClickMock).toHaveBeenCalledTimes(1)
  })
})
