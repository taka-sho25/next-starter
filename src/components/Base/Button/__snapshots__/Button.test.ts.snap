// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button Component > default snapshot 1`] = `
<div>
  <button
    type="button"
  >
    <div>
      hello
    </div>
  </button>
</div>
`;
